feat/IT-35-Dashboard-page

KinoOfficial opened the pull request:
desktop:
UI:
<img width="1134" alt="image-20250415-093450" src="https://github.com/user-attachments/assets/9bfffcc0-c1cc-4fb6-8378-debe259f5671" />
Actual Implementation:
<img width="1439" alt="image" src="https://github.com/user-attachments/assets/d5a2989c-43a2-4260-a4c9-094de4dab75b" />

mobile page
UI:
<img width="371" alt="image-20250415-093328" src="https://github.com/user-attachments/assets/f88fb898-5f6f-4918-aa17-df8a74f2a748" />
Actual Implementation:

<img width="183" alt="image" src="https://github.com/user-attachments/assets/02090d4e-4978-4111-b124-814166bb10eb" />

Extra:
Implement the function of highlight
Once the button is clicked, it wold be highlighted
<img width="89" alt="image" src="https://github.com/user-attachments/assets/2ce1c0ce-0537-465f-99a5-873422e010a0" />
<img width="563" alt="image" src="https://github.com/user-attachments/assets/b381965c-9c97-4924-8cbc-7df036d6990b" />

Implement the dynamic effect
Once the device is too high, the component change the height to fill the blank
<img width="513" alt="image" src="https://github.com/user-attachments/assets/41e5b811-42b6-4307-a702-cdf72d77d93b" />

Implement the reusability of components(sidebar and footer)
<img width="611" alt="image" src="https://github.com/user-attachments/assets/04a3b594-8bc7-42b4-822a-d4a0ad48fcc0" />

Chris-Wang commented:
<img width="326" alt="image" src="https://github.com/user-attachments/assets/83d72a14-8d5c-46d4-b60e-98e28d715d50" />
bottom bar overlap the Go pro card

Chris-Wang commented:
Next button is still there

Chris-Wang commented:
Go pro card is toooooo big
<img width="759" alt="image" src="https://github.com/user-attachments/assets/a7b69456-d0c7-498f-bf80-70bba950f48a" />

Chris-Wang commented on src/app/dashboard/scan/page.tsx:
scan is at the same level as dashboard page

KinoOfficial replied:
Resolved

Chris-Wang commented on src/app/dashboard/home/page.tsx:
```diff
@@ -0,0 +1,19 @@
+'use client';
```
we don't need to have a page like dashboard/home, dashboard is like the home page for user after sign in

KinoOfficial replied:
Resolved

Chris-Wang commented on src/app/module/dashboard/home/desktopHomePage.tsx:
```diff
@@ -0,0 +1,87 @@
+            </Text>
+          </Flex>
+        </Box>
+
```
remove empty lines

Chris-Wang commented on src/app/module/dashboard/home/desktopHomePage.tsx:
```diff
@@ -0,0 +1,87 @@
+            <FeatureCard key={index} {...feature} />
+          ))}
+        </Flex>
+
```
remove empty lines

KinoOfficial replied:
Resolved

Chris-Wang commented on src/app/module/dashboard/sidebar.tsx:
```diff
@@ -0,0 +1,107 @@
+          <Image src="/logo.png"></Image>
+        </Box>
+      </Flex>
+
```
remove empty lines

Chris-Wang commented on src/app/module/dashboard/sidebar.tsx:
```diff
@@ -0,0 +1,107 @@
+          />
+        ))}
+      </Flex>
+
```
remove empty lines

KinoOfficial replied:
Resolved

Chris-Wang commented on src/app/module/dashboard/home/FeatureCard.tsx:
```diff
@@ -0,0 +1,114 @@
+      >
+        {title}
+      </Text>
+
```
remove empty line

KinoOfficial replied:
Resolved

Chris-Wang commented on src/app/module/dashboard/home/FeatureCard.tsx:
```diff
@@ -0,0 +1,114 @@
+          zIndex="1"
+        />
+      )}
+
```
remove empty line

KinoOfficial replied:
Resolved

Chris-Wang reviewed (changes requested):
Please resolve comments

Chris-Wang reviewed (changes requested):
need squash your comments

Chris-Wang commented on src/app/module/sign-in-page/sign-in-form.tsx:
```diff
@@ -26,7 +25,7 @@ export const SignInForm = () => {
   });
 
   const handleGoogleOauth = () => {
-    window.location.href = `${API_BASE_URL}/users/googleAuth`;
```
please use env variables not direct url

Chris-Wang commented on src/app/module/sign-in-page/sign-in-form.tsx:
```diff
@@ -37,7 +36,7 @@ export const SignInForm = () => {
 
       // TODO: Replace with actual API call
-      const response = await axios.post(`${API_BASE_URL}/users/login`, {
+      const response = await axios.post('http://localhost:8000/api/v1/users/login', {
```
please use env variables not direct url

Chris-Wang commented on src/app/module/sign-up-page/sign-up-form.tsx:
```diff
@@ -39,7 +38,7 @@ export const SignUpForm = () => {
   // Handle Google OAuth signup
   const handleGoogleOauth = () => {
-    window.location.href = `${API_BASE_URL}/users/googleAuth`;
+    window.location.href = 'http://localhost:8000/api/v1/users/googleAuth';
```
please use env variables not direct url

Chris-Wang commented on src/app/module/sign-up-page/sign-up-form.tsx:
```diff
@@ -50,7 +49,7 @@ export const SignUpForm = () => {
 
       // TODO: Replace with actual API call
-      await axios.post(`${API_BASE_URL}/users/register`, {
+      await axios.post('http://localhost:8000/api/v1/users/register', {
```
please use env variables not direct url

Chris-Wang commented on src/app/link/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        bgSize="cover"
+      >
+        <Sidebar />
+        <div>This is link page，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/link/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        <div>This is link page，WElcome {user ? user.name : 'Guest'}</div>
+      </Flex>
+      <Box display={{ base: 'block', md: 'none' }} px="20px">
+        <div>This is link page，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/module/dashboard/home/desktopHomePage.tsx:
```diff
@@ -0,0 +1,87 @@
+            </Flex>
+          </Box>
+        </Box>
+
```
remove empty line

Chris-Wang commented on src/app/module/dashboard/home/mobileHomePage.tsx:
```diff
@@ -0,0 +1,180 @@
+    startHeight = '150px';
+  }
+  if (!user) {
+    return <div>Loading...</div>;
```
use animation instead

Chris-Wang commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        bgSize="cover"
+      >
+        <Sidebar />
+        <div>This is ProfilePage ，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        <div>This is ProfilePage ，WElcome {user ? user.name : 'Guest'}</div>
+      </Flex>
+      <Box display={{ base: 'block', md: 'none' }} px="20px">
+        <div>This is ProfilePage ，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/scan/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        bgSize="cover"
+      >
+        <Sidebar />
+        <div>This is scan page，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/scan/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        <div>This is scan page，WElcome {user ? user.name : 'Guest'}</div>
+      </Flex>
+      <Box display={{ base: 'block', md: 'none' }} px="20px">
+        <div>This is scan page，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/toolbox/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        bgSize="cover"
+      >
+        <Sidebar />
+        <div>This is ToolBoxPage ，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/toolbox/page.tsx:
```diff
@@ -0,0 +1,26 @@
+        <div>This is ToolBoxPage ，WElcome {user ? user.name : 'Guest'}</div>
+      </Flex>
+      <Box display={{ base: 'block', md: 'none' }} px="20px">
+        <div>This is ToolBoxPage ，WElcome {user ? user.name : 'Guest'}</div>
```
WElcome?

Chris-Wang commented on src/app/login-callback/page.tsx:
```diff
@@ -1,38 +1,20 @@
       router.push(`/auth-error-page`);
     }
-  }, [failAuth, router]);
+  }, [failAuth]);
```
this will trigger a missing dependency warning

Chris-Wang commented on src/app/login-callback/page.tsx:
```diff
@@ -55,7 +37,19 @@ export default function AuthCallBack() {
       }
     }
-  }, [router, setErrorMessage, validateToken]);
+  }, [router, setErrorMessage]);
```
this will trigger a missing dependency warning

Chris-Wang commented on src/app/dashboard/layout.tsx:
```diff
@@ -0,0 +1,68 @@
+      }
+    };
+    fetchUserData();
+  }, [token]);
```
this will trigger a missing dependency warning

Chris-Wang reviewed (changes requested):
squash your commits, and ask some member to test your branch

StevenRen929 commented:
![IMG_3066](https://github.com/user-attachments/assets/7fba0f99-9bab-4a46-80e4-8862f70f2fe4)
![IMG_3065](https://github.com/user-attachments/assets/da7685f7-fb1d-49be-8d25-5c38b9e6f966)
To ensure the project have a tidy git history,PR will be closed if you don’t follow the pr requirements discussed at the beginning of the project.It happens on IT-35 more than 2 times (with 2,3,4 …. commit nodes). Please squash them ASAP(within an hour).

Chris-Wang committed:
feat(IT-35): complete dashboard page
Added dashboard page, setup user context, added footer and sidebar.

Resolve IT-35
